maplibre-navigation-core/src/androidUnitTest/kotlin/org/maplibre/navigation/core/navigation/MapLibreNavigationTest.kt

```diff
@@ -1,35 +1,51 @@
-package org.maplibre.navigation.android.navigation.v5.navigation
+package org.maplibre.navigation.core.navigation
 
-import android.content.Context
-import io.mockk.every
 import io.mockk.mockk
 import io.mockk.verify
-import org.junit.Assert
-import org.junit.Test
-import org.maplibre.android.location.engine.LocationEngine
-import org.maplibre.navigation.android.navigation.v5.BaseTest
-import org.maplibre.navigation.android.navigation.v5.milestone.BannerInstructionMilestone
-import org.maplibre.navigation.android.navigation.v5.milestone.Milestone
-import org.maplibre.navigation.android.navigation.v5.milestone.StepMilestone
-import org.maplibre.navigation.android.navigation.v5.milestone.VoiceInstructionMilestone
-import org.maplibre.navigation.android.navigation.v5.offroute.OffRoute
-import org.maplibre.navigation.android.navigation.v5.snap.Snap
-import org.maplibre.navigation.android.navigation.v5.snap.SnapToRoute
-
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.test.setMain
+import org.junit.Before
+import org.maplibre.navigation.core.BaseTest
+import org.maplibre.navigation.core.location.engine.LocationEngine
+import org.maplibre.navigation.core.milestone.BannerInstructionMilestone
+import org.maplibre.navigation.core.milestone.Milestone
+import org.maplibre.navigation.core.milestone.StepMilestone
+import org.maplibre.navigation.core.milestone.VoiceInstructionMilestone
+import org.maplibre.navigation.core.offroute.OffRoute
+import org.maplibre.navigation.core.snap.Snap
+import org.maplibre.navigation.core.snap.SnapToRoute
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import kotlin.test.assertNotNull
+import kotlin.test.assertEquals
+import kotlin.test.assertNotSame
+
+@OptIn(ExperimentalCoroutinesApi::class, DelicateCoroutinesApi::class)
 class MapLibreNavigationTest : BaseTest() {
+
+    private val mainThreadSurrogate = newSingleThreadContext("UI thread")
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(mainThreadSurrogate)
```

Comment:
Consider resetting the main dispatcher in a teardown to avoid polluting other test classes.